### ISSUE TITLE
[BUG]: PullRequest's `$.pull_request.{base,head}.repo` does not have `custom_properties` field

### ISSUE THREAD

**Issue by w568w** [octokit/webhooks#911]
### What happened?

Currently `$.pull_request.{base,head}.repo` in Pull Request-related events (e.g. `pull_request reopened event`) is marked as `#/definitions/repository`, which has a mandatory `custom_properties` field. 

It is a mistake because in GitHub docs, [`pull_request` events do not have such a field](https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=reopened#pull_request) there. And this schema will not pass validation with a strict validator.

It is strange because #900 reported the opposite behavior recently. I suppose there might have been an API regression from GitHub side.

[Here](https://0x0.st/Xrfp.json) is an example payload.

### Versions

`@octokit/webhooks-schemas@7.4.0` with [`octokit-rs`](https://github.com/miaxos/octokit-rs).

### Relevant log output

```shell
`Error("missing field `custom_properties`", line: 195, column: 17)`
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wolfy1339** [comment]
That's odd that one event has the property for `repository` and another doesn't.

It's usually always present

**Comment by w568w** [comment]
Yes, I believe it's GitHub that should be responsible for this issue, but so far I haven't seen any public issue reports, regression tracking, or changelogs mentioning the removal of this field.

I'm not sure when this will be fixed. But for the sake of consistency with the official documentation (and to not break the workflow for all those using the strict validator), I think it's necessary to fix this for now, temporarily, in terms of the schema definition.

**Comment by wolfy1339** [comment]
I cross-checked with the OpenAPI spec, and indeed this problem is not present there.

The best thing to do is add a new repository type for PRs

**Comment by w568w** [comment]
Thanks for taking the time to notify the downstream repository.

> I cross-checked with the OpenAPI spec, and indeed this problem is not present there.

Do you mean the OpenAPI spec has no such problem? I checked <https://github.com/octokit/openapi-webhooks/releases/download/v8.2.0/api.github.com.json>: its `#/components/schemas/webhook-pull-request-reopened` refers to `#/components/schemas/repository-webhooks`, which does have `custom_properties` property.

**Comment by wolfy1339** [comment]
Please see https://github.com/octokit/webhooks/issues/901

This repository is woefully out of date and not maintained anymore. It's hand crafted schemas based on payload examples and GitHub documentation from before the release of the OpenAPI schema for webhooks.
The update process that was used in this repository is broken by the switch to OpenAPI for the documentation.

Please switch to using the OpenAPI spec that is located at https://github.com/octokit/openapi-webhooks/

**Comment by w568w** [comment]
@wolfy1339 Thanks for the friendly heads-up (two years later)!

Given the potential compatibility issues, I've stopped using the schema library for long and now parse the JSON fields manually instead. As it turns out, the latter may be more stable and less likely to break with API updates.

Anyway, I'll close this irrevelant issue.
